clang-scan-deps ignores warning supressions

With the following file:

```cpp
module;

export module mylib;

#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Winclude-angled-in-module-purview"
#include <header.hpp> // header contents irrelevant
```

And the following build command:

```sh
#!/bin/bash

set -e

SRCDIR="$(cd "$(dirname "$0")" && pwd)"

clang-scan-deps-22 \
    -- \
    clang++-22 \
    -I"$SRCDIR" \
    -stdlib=libc++ \
    -g \
    -std=c++23 \
    -x c++ \
    "$SRCDIR/mylib.cpp" \
    -c -o mylib.cpp.o 
```

The pragma seems to be ignored:

```
Diagnostics while scanning dependencies for '[redacted]/modules-pg2/mylib.cpp':
[redacted]/modules-pg2/mylib.cpp:8:10: warning: '#include <filename>' attaches the declarations to the named module 'mylib', which is not usually intended; consider moving that directive before the module declaration
mylib.cpp.o: [redacted]/modules-pg2/mylib.cpp \
  [redacted]/modules-pg2/header.hpp
```

The compiler does not emit the diagnostic. It is only the scanner. This creates a cluttered log when legitimately including files in the purview to create module bindings.

--- a/clang/lib/DependencyScanning/DependencyScannerImpl.cpp
+++ b/clang/lib/DependencyScanning/DependencyScannerImpl.cpp
@@ -412,6 +412,8 @@ void dependencies::initializeScanCompilerInstance(
   ScanInstance.setBuildingModule(false);
   ScanInstance.createVirtualFileSystem(FS, DiagConsumer);
   ScanInstance.createDiagnostics(DiagConsumer, /*ShouldOwnClient=*/false);
+  if (Service.getOpts().Format == ScanningOutputFormat::P1689)
+    ScanInstance.getDiagnostics().setIgnoreAllWarnings(true);
   ScanInstance.createFileManager();
   ScanInstance.createSourceManager();
 

--- a/clang/test/ClangScanDeps/p1689-suppress-warnings.cppm
+++ b/clang/test/ClangScanDeps/p1689-suppress-warnings.cppm
@@ -1,0 +1,23 @@
+// Test that P1689 format suppresses warnings during dependency scanning.
+// This addresses https://github.com/llvm/llvm-project/issues/185394
+
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: split-file %s %t
+
+// Test P1689 format - should NOT emit warnings
+// RUN: clang-scan-deps -format=p1689 -- %clang++ -std=c++20 -I%t -c %t/mylib.cppm -o %t/mylib.o 2>&1 | FileCheck %s
+
+// CHECK-NOT: warning:
+// CHECK: {
+// CHECK: "revision"
+
+//--- header.h
+// Empty header for testing
+
+//--- mylib.cppm
+module;
+
+export module mylib;
+
+#include <header.h>

